api/v1alpha1/ext_auth_types.go

```diff
@@ -63,6 +63,65 @@ type ExtAuth struct {
 	//
 	// +optional
 	RecomputeRoute *bool `json:"recomputeRoute,omitempty"`
+
+	// ContextExtensions are analogous to http_request.headers, however these
+	// contents will not be sent to the upstream server. This provides an
+	// extension mechanism for sending additional information to the auth server
+	// without modifying the proto definition. It maps to the internal opaque
+	// context in the filter chain.
+	//
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=name
+	ContextExtensions []*ContextExtension `json:"contextExtensions,omitempty"`
+}
+
+// ContextExtensionValueType defines the types of values for ContextExtension supported by Envoy Gateway.
+//
+// +kubebuilder:validation:Enum=Value;ValueRef
+type ContextExtensionValueType string
+
+const (
+	// ContextExtensionValueTypeValue defines the "Value" ContextExtension type.
+	ContextExtensionValueTypeValue ContextExtensionValueType = "Value"
+
+	// ContextExtensionValueTypeValueRef defines the "ValueRef" ContextExtension type.
+	ContextExtensionValueTypeValueRef ContextExtensionValueType = "ValueRef"
+)
+
+// ContextExtension is analogous to http_request.headers, however these
+// contents will not be sent to the upstream server. This provides an
+// extension mechanism for sending additional information to the auth server
+// without modifying the proto definition. It maps to the internal opaque
+// context in the filter chain.
```

Comment:
> It maps to the internal opaque context in the filter chain.

Is this phrasing needed? It is a little confusing, as there are no filter chain modifications involved in this implementation. I think you mean:

"The values provided here are mapped into ContextExtensions in the CheckRequest that external authorization server receives"

Reply:
It is a copy/paste from the Envoy documentation, I did not want to re-invent the wheel regarding how it is documented

https://pkg.go.dev/github.com/envoyproxy/go-control-plane/envoy/service/auth/v3#AttributeContext.ContextExtensions

Reply:
Fair enough. I don't agree with the go-control-plane docs, but let's keep it as-is for consistency.